src/backend/distributed/executor/citus_custom_scan.c

```diff
@@ -190,6 +194,28 @@ CitusBeginScan(CustomScanState *node, EState *estate, int eflags)
 		CitusBeginModifyScan(node, estate, eflags);
 	}
 
+	Job *workerJob = scanState->distributedPlan->workerJob;
+
+	/*
+	 * If the Job has the partition key, check for a match with the force_pushdown
+	 * distribution argument.
+	 */
+	if (workerJob->partitionKeyValue)
```

Comment:
should this be `AllowedDistributionColumnValue != NULL`?

Reply:
No, if a Job (such as Update) doesn't have a key there is no need to compare.

Reply:
For only FastPathQueries, we have the correct shard-key at the top of JobTree. For complex queries, such as sub-query, we have the Job-shardkey as NULL. The compare was failing prematurely by comparing NULL with distribution argument.

Reply:
Hmz, but that seems much less restrictive than what we had at an earlier stage.

For instance, this is allowed to go through on a single node, and then later breaks with scale out:

```sql
CREATE OR REPLACE FUNCTION test(p_x int, p_y int)
RETURNS bigint
AS $$
DECLARE
    res bigint;
BEGIN
    INSERT INTO test VALUES (p_x, p_y);
    DELETi FROM test;
    RETURN p_y;
END;
$$ LANGUAGE plpgsql;
SELECT create_distributed_function('test(int,int)','p_x',force_pushdown :=true);


CREATE OR REPLACE FUNCTION my_test(x int, y int)
RETURNS void
AS $$
DECLARE
    v int;
BEGIN
    PERFORM 1 FROM test(x, y);
    PERFORM 1 FROM test(x + 1, y);
END;
$$ LANGUAGE plpgsql;

SELECT my_test(1,10);
```

Reply:
That's how the code used to behave, but it has its own issues, I reverted it now. Please check.